detector.py

```diff
@@ -49,30 +55,232 @@ def detect(self, image, camera: Camera) -> List[PersonView]:
+    https://github.com/CMU-Perceptual-Computing-Lab/openpose/releases
+    """
+    class Models(Enumeration):
+        coco = 'COCO'
+        body_25 = 'BODY_25'
+
+    models = Models()  # enum for OP models
+
+    def __init__(self, binary_path, using_gpu, net_resolution='-1x240', force_op_model=None):
+        """
+        1) Go to OpenPose releases: https://github.com/CMU-Perceptual-Computing-Lab/openpose/releases
+        2) Download and extract OpenPose folder (referred as OP_HOME).
+        3) Run OP_HOME/models/getModels.bat to download all OP models.
+            [OPTIONAL] edit the getModels.bat to download only needed models.
+        4) [OPTIONAL] Read https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/quick_start.md
+        :param binary_path: path to OpenPose binary
+        :param using_gpu: True to use GPU, False to use CPU. To use GTX GPU, it should support OpenCL 1.2. Install CUDA
+            drivers from: https://developer.nvidia.com/cuda-downloads
+            Then follow cuDNN installation guide: https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html
+        :param net_resolution: Multiples of 16, e.g. 320x176. Increase ~ accuracy increase. Decrease ~ speed increase.
+            For best results, keep the closest aspect ratio possible to the images processed. Using -1 in any of the
+            dimensions, OP will choose the optimal aspect ratio depending on the input. E.g. the default -1x368 is
+            equivalent to 656x368 for 16:9 resolutions (full HD 1980x1080, HD 1280x720 etc.). NOTE: Higher resolution
+            means higher memory consumption (RAM or GPU memory). E.g. -1x336 (for full HD) takes 2 GB of GTX 760 memory.
+        :param force_op_model: Manually select OpenPose model BODY_25/COCO. By default, OP model is BODY_25 for GPU,
+        COCO for CPU. Because COCO is ~3x faster on CPU than BODY_25, but BODY_25 is ~40% faster on GPU.
+            Make sure to have the requested model downloaded in OP_HOME/models folder.
+        """
         if 'Windows' not in platform.system():
             raise NotImplementedError("Only Windows binaries supported.")
 
-        assert os.path.isfile(binary_path), 'OpenPose binary not found. Path: {}'.format(binary_path)
+        assert os.path.isfile(binary_path), "OpenPose binary not found. Path: {}".format(binary_path)
+        self.binary_home = os.path.dirname(binary_path).rstrip('bin')  # OpenPoseDemo.exe is in bin/ subdirectory
+
+        if force_op_model and not self.models.contains(force_op_model):
+            logger.error("Using an unknown OpenPose model: {0}. Select from {1}.models: {2}!"
+                         .format(force_op_model, type(self).__name__,  self.models.enum_values()))
+            raise NotImplementedError("Using unknown OpenPose model!")
+
+        # prepare tmp directory for input images and results; tmp dir is deleted in obj destructor
+        tmp_dir_name = 'POVa_pedestrian_tracking_TEMP_DIR'
```

Comment:
Function [tempfile.TemporaryDirectory](https://docs.python.org/3.7/library/tempfile.html#tempfile.TemporaryDirectory) could be used as a context manager instead.

Reply:
This would need to create temp file for every call of detection. It is not preferred here, plus now user can check results of previous detection (without disappearing). I agree that TemporaryDirectory is the "correct" way, but it was convenient to debug detection this way :D
PS: If there is a way to make TemporaryDirectory persistent, I didn't know it and I'm sorry :D